subprojects/delivery/signer/src/main/kotlin/com/avito/plugin/AbstractSignTask.kt

```diff
@@ -0,0 +1,104 @@
+package com.avito.plugin
+
+import com.avito.android.Problem
+import com.avito.android.stats.statsd
+import com.avito.gradle.worker.inMemoryWork
+import com.avito.http.HttpClientProvider
+import com.avito.http.RetryInterceptor
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+import com.avito.plugin.internal.SignViaServiceAction
+import com.avito.time.DefaultTimeProvider
+import com.avito.time.TimeProvider
+import com.avito.utils.buildFailer
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkerExecutor
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+public abstract class AbstractSignTask(private val workerExecutor: WorkerExecutor) : DefaultTask() {
+
+    @get:Input
+    public abstract val tokenProperty: Property<String>
+
+    @get:Input
+    public abstract val serviceUrl: Property<String>
+
+    @get:Input
+    public abstract val readWriteTimeoutSec: Property<Long>
+
+    @get:Internal
+    public abstract val loggerFactory: Property<LoggerFactory>
+
+    @get:OutputFile
+    public abstract val signedArtifact: Property<RegularFile>
+
+    protected abstract fun unsignedFile(): File
+
+    @TaskAction
+    public fun doWork() {
+        val loggerFactory = loggerFactory.get()
+        val unsignedFile = unsignedFile()
+        val signedFile = signedArtifact.get().asFile
+        val timeProvider: TimeProvider = DefaultTimeProvider()
+        val serviceUrl: HttpUrl = serviceUrl.map { it.toHttpUrl() }.get()
+
+        val timeout = readWriteTimeoutSec.get()
+
+        val httpClient = HttpClientProvider(
+            statsDSender = project.statsd.get(),
+            timeProvider = timeProvider,
+            loggerFactory = loggerFactory
+        ).provide()
+            .writeTimeout(timeout, TimeUnit.SECONDS)
+            .readTimeout(timeout, TimeUnit.SECONDS)
+            .addInterceptor(
+                RetryInterceptor(
+                    retries = 3,
+                    allowedMethods = listOf("GET", "POST")
+                )
+            )
+            .build()
+
+        val logger = loggerFactory.create<LegacySignArtifactTask>()
+
+        workerExecutor.inMemoryWork {
+            SignViaServiceAction(
+                serviceUrl = serviceUrl,
+                httpClient = httpClient,
+                token = tokenProperty.get(),
```

Comment:
Propose to unwrap directly in the task action method
Is the error message is clear?

Reply:
Good catch!
That's the lack of tests showing up
will cover that scenario